knimin/lib/data_access.py

```diff
@@ -2399,6 +2399,134 @@ def getProjectNames(self):
         sql = """SELECT project FROM project"""
         return [x[0] for x in self._con.execute_fetchall(sql)]
 
+    def get_plate_info(self, plate_id):
+        """Gets attributes of a plate by id
+
+        Parameters
+        ----------
+        plate_id : int
+            Plate id
+
+        Returns
+        -------
+        dict
+            Plate attributes
+        """
+        sql = """SELECT * FROM pm.plate WHERE plate_id = %s"""
+        return self._con.execute_fetchdict(sql, [plate_id])[0]
+
+    def set_plate_info(self, plate_id, plate_info):
+        """Sets attributes of a plate by id
+
+        Parameters
+        ----------
+        plate_id : int
+            Existing plate id (>0) or create new plate (0)
+        plate_info : dict
+            Plate attributes
+
+        Returns
+        -------
+        int
+            Plate id of the created / modified plate
+        """
+        keys = ', '.join([str(x) for x in plate_info.keys()])
+        values = ', '.join(['\'' + str(x) + '\'' for x in plate_info.values()])
+        sql = ''
+        if plate_id == 0:  # Create new plate
+            sql = 'INSERT INTO pm.plate (' + keys + ') VALUES (' + values + ')'
```

Comment:
The column names are not supported to be passed through parameters, only the values. That is intended by the psycopg2 developers. The way of doing this specific call will be something along the lines of:
```python
sql = "INSERT INTO pm.plate ({}) VALUES ({})".format(keys, ','.join(['%s'] * len(plate_info)))
self._con.execute_fetchone(sql, values)
```

This way psycopg2 will do the conversion for you - and will be safer.